// This proto file defines the objects that is responsible for various
// functionalities related to library

syntax = "proto3";

import "google/protobuf/any.proto";

package inventory;

// The `Status` type defines a logical error model.
message ResponseStatus {

  // A simple error code that can be easily handled by the client. The
  // actual error code is defined by `google.rpc.Code`.
  optional int32 code = 1;

  // A developer-facing human-readable error message in English.
  // Explains the error and offer an actionable resolution to it.
  optional string message = 2;

  // Additional error information that the client code can use to handle
  // the error, such as retry info or a help link.
  repeated google.protobuf.Any details = 3;
}

message Book {
  optional string isbn = 1;
  optional string title = 2;
  optional string author = 3;

  enum Genre {
    FICTION_THRILLER = 0; //default
    FICTION_ROMANTIC = 1;
    PHILOSOPHY = 2;
    HISTORY = 3;
    FICTION_ADVENTURE = 4;
  }
  optional Genre genre = 4;
  optional int32 pub_year = 5;
}

//Can be anything: CD, Books, EBooks, Magazines etc. Right now its just Book
message InventoryItem {
  optional int32 inventory_number = 1;
  oneof item {
        Book book = 5;
  }

  enum IssueStatus {
    AVAILABLE = 0; //default
    TAKEN = 1;
  }
  optional IssueStatus status = 2;
}

